pkg/timeutil/timeutil.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 clair authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package timeutil implements extra utilities dealing with time not found
+// in the standard library.
+package timeutil
+
+import (
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/coreos/clair/pkg/stopper"
+	log "github.com/sirupsen/logrus"
+)
+
+// ApproxSleep is a stoppable time.Sleep that adds a slight random variation to
+// the wakeup time in order to prevent thundering herds.
+func ApproxSleep(approxWakeup time.Time, st *stopper.Stopper) (stopped bool) {
+	waitUntil := approxWakeup.Add(time.Duration(rand.ExpFloat64()/0.5) * time.Second)
```

Comment:
nit: should we make this random variable configurable?

Reply:
It's pretty common to not have it be configurable. If the signature was sleeping for a duration rather than sleeping until a specific time, I'd agree with you, though.